packages/phone-number-privacy/signer/test/integration/pnp.test.ts

```diff
@@ -1003,23 +1037,19 @@ describe('pnp', () => {
           const authorization = getPnpRequestAuthorization(req, PRIVATE_KEY1)
           const res = await sendRequest(req, authorization, SignerEndpoint.PNP_SIGN)
 
-          expect(res.status).toBe(200)
-          expect(res.body).toMatchObject<SignMessageResponseSuccess>({
-            success: true,
+          expect(res.status).toBe(500)
+          expect(res.body).toStrictEqual<SignMessageResponseFailure>({
+            success: false,
             version: res.body.version,
-            signature: expectedSignature,
-            performedQueryCount: performedQueryCount + 1,
-            totalQuota: expectedQuota,
-            blockNumber: testBlockNumber,
-            warnings: [ErrorMessage.FAILURE_TO_STORE_REQUEST],
+            error: ErrorMessage.UNKNOWN_ERROR,
```

Comment:
Can definitely add a TODO and agree that it isn't necessary pre-merge.

My thinking on this during the PR:
- the most "straight forward" seeming way (adding try/catch's to each block of code that we want to return specific errors) causes the code to become more complicated (nervous that we will very likely add in some weird bugs/edge cases in the future since we would again need to add in rollbacks for each of these errors and return)
- we could try to match on certain types of error messages in the outer handler (i.e. DB, key errors) and return specific responses; I started looking into this but then decided it would be a bit out of scope for this PR (it felt a bit more involved than it sounds as soon as I started actually implementing it)
- we could do something kinda weird like put all of the code that needs to rollback in the last try/catch and have a "most likely error" param that changes to reference the coming block and then handle the errors + rollback accordingly (though this is also pretty weird)

Created a ticket here: https://github.com/celo-org/celo-monorepo/issues/9882